### PR TITLE
Support ASR Interface v1.6

### PR DESCRIPTION
**Pull request by shshshl09**
* apply listenTimeoutFailBeep

### CODE REVIEW

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/DefaultASRAgent.kt [hunk 1]
```diff
@@ -722,17 +726,23 @@ class DefaultASRAgent(
...
 
+                    val payload = param.expectSpeechDirectiveParam?.directive?.payload
                     if (type == ASRAgentInterface.ErrorType.ERROR_RESPONSE_TIMEOUT) {
-                        sendResponseTimeout(param.expectSpeechDirectiveParam?.directive?.payload, header.dialogRequestId)
+                        sendResponseTimeout(payload, header.dialogRequestId)
+                        speechToTextConverterEventObserver.onError(type, header)
                     } else if (type == ASRAgentInterface.ErrorType.ERROR_LISTENING_TIMEOUT) {
-                        sendListenTimeout(param.expectSpeechDirectiveParam?.directive?.payload, header.dialogRequestId)
+                        sendListenTimeout(payload, header.dialogRequestId)
+                        speechToTextConverterEventObserver.onError(type, header, payload?.listenTimeoutFailBeep != false)
```
Comment (Yuseon-Han) [inline nugu-agent/src/main/java/com/skt/nugu/sdk/agent/DefaultASRAgent.kt:744]: if payload is null "payload?.listenTimeoutFailBeep != false" returns true.
is it your intent?
I think you was trying to mean "payload?.listenTimeoutFailBeep == true".
I think  "payload?.listenTimeoutFailBeep == true" is appropriate.

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ASRAgentInterface.kt [hunk 2]
```diff
@@ -159,27 +159,33 @@ interface ASRAgentInterface {
...
          */
         fun onCompleteResult(result: String, header: Header)
         /**
          * Called when occur error on recognizing.
          * @param type reason for error
+         * @param header the request header
+         * @param allowEffectBeep whether allow beep play or not
          */
-        fun onError(type: ErrorType, header: Header)
+        fun onError(type: ErrorType, header: Header, allowEffectBeep: Boolean = true)
```
Comment (Yuseon-Han) [inline nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ASRAgentInterface.kt:183]: allowEffectBeep is for TimeoutError. not all Error. 
I think this must be commented and variable name must be changed to contain the meaning of 'timeout'.

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ExpectSpeechPayload.kt [hunk 3]
```diff
@@ -94,6 +97,7 @@ data class ExpectSpeechPayload private constructor(
         result = 31 * result + (domainTypes?.contentHashCode() ?: 0)
         result = 31 * result + (asrContext?.hashCode() ?: 0)
         result = 31 * result + (epd?.hashCode() ?: 0)
+        result = 31 * result + (listenTimeoutFailBeep?.hashCode() ?: 0)
```
Comment (Yuseon-Han) [inline nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ExpectSpeechPayload.kt:100]: kotlin data class automatically generate equals(), and hashCode(). 
override is not need.
check it later.

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/DefaultASRAgent.kt (hunk 1 above)
Comment (shshshl09) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/DefaultASRAgent.kt:744]: yes, default setting is true.

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ExpectSpeechPayload.kt (hunk 3 above)
Comment (shshshl09) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ExpectSpeechPayload.kt:100]: re-define recommended due to following array field.
=> val domainTypes: Array<String>?,

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ASRAgentInterface.kt (hunk 2 above)
Comment (shshshl09) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/asr/ASRAgentInterface.kt:183]: If we need a beep field for another type later, we need to add another param, so name it generally.

**Review by Yuseon-Han — APPROVED** [review]

**Pull request merged**